File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int num = n;
        int sum = 0;
        int prod = 1;
        while(n > 0){
            int digit = n % 10;
            n /= 10;
            sum += digit;
            prod *= digit;
        }
        int SUM = sum + prod;
        return num % SUM == 0;
    }
};

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        num = n
        total = 0
        prod = 1
        while n > 0:
            digit = n % 10
            n //= 10
            total += digit
            prod *= digit
        SUM = total + prod
        return num % SUM == 0